python/pydantic_core/core_schema.py

```diff
@@ -659,6 +660,72 @@ def float_schema(
     )
 
 
+class DecimalSchema(TypedDict, total=False):
+    type: Required[Literal['decimal']]
+    gt: Union[int, Decimal]
+    ge: Union[int, Decimal]
+    lt: Union[int, Decimal]
+    le: Union[int, Decimal]
+    max_digits: int
+    decimal_places: int
+    multiple_of: Union[int, Decimal]
```

Comment:
Should `float` be allowed here? I think we allow that with some tolerance elsewhere

Reply:
Actually because `lax_decimal` will accept `int` and `float` there is no need for `Union` here, this should just be `Decimal`.